backend/src/app.controller.ts

```diff
@@ -16,38 +17,30 @@ export class AppController {
   /**
    * Returns a list the ids of all nodes
    */
-  @Get('getAllNodes')
-  async getAllNodes(): Promise<number[]> {
-    return this.appService.getAllNodes();
+  @Post('queryAll')
+  async queryAll(@Body() query?: LimitQuery): Promise<QueryResult> {
+    return this.appService.queryAll(query);
```

Comment:
Why do we even need this `AppService` type any longer, when the controller does noting but forward the call to it and the service does all the heavy lifting?

Reply:
Controller: Handles requests (validates params, auth, ...).
Service: Does the "logic". Fully functional without any controller.

So a controller is the interface to the outside world (physical connection), a service does the inner logic without any information about the origin of the query (i.e. was it sent via REST or via the command line).